cookbooks/wordpress/resources/theme.rb

```diff
@@ -52,6 +52,21 @@
         user node[:wordpress][:user]
         group node[:wordpress][:group]
       end
+    elsif theme_repository.end_with?(".zip")
+      zip_path = "#{Chef::Config[:file_cache_path]}/#{new_resource.theme}.zip"
+
+      remote_file zip_path do
+        source theme_repository
+        action :create
+      end
+
+      archive_file zip_path do
+        destination theme_directory
+        action :extract
```

Comment:
This should be `action :nothing` and then subscribe to changes in the zip with `subscribes :extract, "remote_file[#{zip_path}]` or it will wind up being unpacked every time chef runs.

Reply:
That's much better indeed.